### PR TITLE
add abi_types unit tests

### PR DESCRIPTION
**Pull request by iFrostizz**
### What I did

Filling some coverage holes regarding abi types that cannot be hit with integration testing because the type checker would have, well, caught it

### How I did it

Adding a parametrized unit test

### Commit message

add abi types unit tests

### Description for the changelog

add abi types unit tests

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by iFrostizz** [comment]
Note: this is about https://github.com/vyperlang/vyper/issues/3647 but won't probably make it reach 90%

File: tests/parser/types/test_bytes.py [hunk 1]
```diff
@@ -333,3 +334,61 @@ def assign():
...
+
+cases_invalid_types = [
+    (
+        ABI_GIntM,
+        [
+            (0, False),
+            (7, False),
+            (300, True),
+            (300, False)
+        ],
```
Comment (charles-cooper) [inline tests/parser/types/test_bytes.py:346]: nit: these args could be a tuple

File: tests/parser/types/test_bytes.py [hunk 2]
```diff
@@ -333,3 +334,61 @@ def assign():
...
+        CompilerPanic
+    ),
+]
+
+# double parametrization cannot work because the 2nd dimension is variable
+cases_invalid_types_flat = []
+for cases in cases_invalid_types:
+    for case in cases[1]:
+        cases_invalid_types_flat.append((cases[0], case, cases[2]))
```
Comment (charles-cooper) [inline tests/parser/types/test_bytes.py:390]: i think it will be more readable to move this loop inside the unit test

**Review by charles-cooper — CHANGES_REQUESTED** [review]
thanks for the test! maybe there is a better place for this test btw? maybe `tests/abi_types/test_invalid_abi_types.py`

File: tests/parser/types/test_bytes.py (hunk 2 above)
Comment (iFrostizz) [reply tests/parser/types/test_bytes.py:390]: Mh, we could do that but the advantage of doing it out of the test was to really parametrize it with these variants

File: tests/parser/types/test_bytes.py (hunk 1 above)
Comment (iFrostizz) [reply tests/parser/types/test_bytes.py:346]: Right !

Comment (iFrostizz) [reply tests/parser/types/test_bytes.py:346]: Thank you :)

**Comment by iFrostizz** [comment]
moved the logic for 2-dimensionnal parametrization inside the test, changed variants to a tuple, and move the file to the new location !

**Comment by iFrostizz** [comment]
as you advised on Discord, because `CompilerPanic` is likely to be the thing that this test tests, it's now not a param anymore

File: vyper/exceptions.py [hunk 3]
```diff
@@ -336,3 +336,7 @@ class UnfoldableNode(VyperInternalException):
 
 class TypeCheckFailure(VyperInternalException):
     """An issue was not caught during type checking that should have been."""
+
+
+class InvalidABIType(VyperInternalException):
+    """Unexpected ABI Type internals"""
```
Comment (charles-cooper) [inline vyper/exceptions.py:342]:
```suggestion
    """An internal routine constructed an invalid ABI type""
```

**Review by charles-cooper — APPROVED** [review]
lgtm, thank you!

**Pull request merged**